test/utils/int/providerRemote.ts

```diff
@@ -20,3 +27,50 @@ export async function clearStoredNotifications(windowIdentity: Identity): Promis
         });
     }, windowIdentity);
 }
+
+/**
+ * Closes all of the service providers child windows and then restarts the main window after a provided amount of time.
+ * Resolves once the service is fully initialized.
+ * This will not reopen any child windows unless the provider automatically does so.
+ */
+export async function restartProvider(snoozeTime: number = 0): Promise<void> {
+    const providerApp = fin.Application.wrapSync(serviceIdentity);
+    const providerAppWindow = await providerApp.getWindow();
+    await providerApp.getChildWindows().then(children => children.forEach(win => win.close()));
+    await providerAppWindow.navigate('about:blank');
+
+    // A small delay is needed in order for OpenFin to handle the navigateBack call successfully.
+    // If a snoozeTime is less than the minimum we enforce the minimum otherwise let it be
+    await delay(snoozeTime <= Duration.NAVIGATE_BACK ? Duration.NAVIGATE_BACK : snoozeTime);
+
+    await providerAppWindow.navigateBack();
+    await providerReady();
+}
+
+/**
+ * Checks that the provider is ready and has been initialized.
+ */
+export async function providerReady(): Promise<void> {
+    const TIMEOUT = 5000;
+    let timedOut = false;
+
+    [timedOut] = await withTimeout(TIMEOUT, new Promise<void>(async (resolve) => {
+        let initialized = false;
+
+        while (!initialized) {
```

Comment:
Would feel safer if this was `while (!initialized && !timedOut)`. Think as is it'll just keep going even after the promise has rejected

Reply:
Good catch. I've made that change.